exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/Otel2PrometheusConverter.java

```diff
@@ -400,29 +403,42 @@ private Exemplars convertDoubleExemplars(List<DoubleExemplarData> exemplars) {
     return Exemplars.of(result);
   }
 
+  @Nullable
   private Exemplar convertExemplar(double value, ExemplarData exemplar) {
     SpanContext spanContext = exemplar.getSpanContext();
+    Labels labels = Labels.EMPTY;
     if (spanContext.isValid()) {
-      return new Exemplar(
-          value,
+      labels =
           convertAttributes(
-              null, // resource attributes are only copied for point's attributes
-              null, // scope attributes are only needed for point's attributes
+              null,
+              null,
               exemplar.getFilteredAttributes(),
               "trace_id",
               spanContext.getTraceId(),
               "span_id",
-              spanContext.getSpanId()),
-          exemplar.getEpochNanos() / NANOS_PER_MILLISECOND);
+              spanContext.getSpanId());
     } else {
-      return new Exemplar(
-          value,
-          convertAttributes(
-              null, // resource attributes are only copied for point's attributes
-              null, // scope attributes are only needed for point's attributes
-              exemplar.getFilteredAttributes()),
-          exemplar.getEpochNanos() / NANOS_PER_MILLISECOND);
+      labels = convertAttributes(null, null, exemplar.getFilteredAttributes());
+    }
+    int runes = getRunes(labels);
+    if (runes > EXEMPLAR_MAX_RUNES) {
+      THROTTLING_LOGGER.log(
+          Level.WARNING,
+          "exemplar labels have " + runes + " runes, exceeding the limit of " + EXEMPLAR_MAX_RUNES);
+      return null;
+    }
+    return new Exemplar(value, labels, exemplar.getEpochNanos() / NANOS_PER_MILLISECOND);
+  }
+
+  private static int getRunes(Labels labels) {
+    int runes = 0;
+    for (Label l : labels) {
+      runes +=
+          new String(l.getName().getBytes(StandardCharsets.UTF_8), StandardCharsets.UTF_8).length()
```

Comment:
you are converting string to a byte array an then back to string again, won't this return the same string that you started with? Isn't this the same as calling `l.getName().length()`?

Reply:
1. From what I understood, the exposition text format follows utf-8 encoding: https://github.com/prometheus/docs/blob/main/content/docs/instrumenting/exposition_formats.md#text-based-format
2. From the golang implementation i observe the length is being calculated using the ut8 api.
So,
First encoding the string into ut8 bytes using `getBytes(StandardCharsets.UTF_8)`
and then creating a new string out of those ut8 encoded bytes. If i omit the charset while creating the new string, it will use the platform's default.
Awaiting further feedback on this.

Reply:
Converting a string to bytes and the back to string will result in a string identical to the initial string, it is not useful.

Reply:
@laurit I might have been wrong in my understanding about calculating 'rune length'. Rune is a character in go. In that case code point in java will correspond to rune in go and so that would simply mean `str.length()` is the requirement here. 
Should i go with `l.getName().length()` regardless of the jvm's default charset, please confirm ?